bionic/deriver.py

```diff
@@ -733,6 +747,16 @@ def _mark_entry_completed(self, completed_entry):
             completed_entry.future = None
             del self._in_progress_entries[completed_entry.state.task.keys[0]]
 
+        if self._is_requested_entry(completed_entry):
```

Comment:
I feel like `_entry_is_requested` reads a little more naturally here. `_is_requested_entry` sounds like we're saying that `self` is a requested entry.

Reply:
But also, what if we just initialized an `entry.is_requested` field when we created each entry?

Reply:
I moved `is_requested` and `results_by_dnode` to `entry` and it looks so much better!